bench/balancer.ts

```diff
@@ -0,0 +1,63 @@
+import chalk from "chalk";
+
+import { OrderBalance, OrderKind } from "../src/ts";
+
+import { BenchFixture } from "./fixture";
+
+async function main() {
+  const fixture = await BenchFixture.create();
+
+  const pad = (x: unknown) => ` ${x} `.padStart(14);
+
+  console.log(chalk.bold("=== Balancer Gas Benchmarks ==="));
+  console.log(
+    chalk.gray(
+      "--------------+--------------+--------------+--------------+--------------",
+    ),
+  );
+  console.log(
+    ["hops", "kind", "sell balance", "buy balance", "gas"]
+      .map((header) => chalk.cyan(pad(header)))
+      .join(chalk.gray("|")),
+  );
+  console.log(
+    chalk.gray(
+      "--------------+--------------+--------------+--------------+--------------",
+    ),
+  );
+
+  for (let hops = 1; hops <= 3; hops++) {
+    for (const kind of [OrderKind.SELL, OrderKind.BUY]) {
+      for (const sellTokenBalance of [
+        OrderBalance.ERC20,
+        OrderBalance.EXTERNAL,
+        OrderBalance.INTERNAL,
+      ]) {
+        for (const buyTokenBalance of [
+          OrderBalance.ERC20,
+          OrderBalance.INTERNAL,
+        ]) {
+          const { gasUsed } = await fixture.swap({
+            hops,
+            kind,
```

Comment:
Given that buy/sell costs are very similar, maybe we can make the table a bit more concise by only using one of the two or averaging both?

Reply:
They do now, but #534 will likely change this as it would allow for a nice optimization for sell orders (but not so much so for buy orders).